localstack/utils/aws/aws_stack.py

```diff
@@ -586,7 +586,15 @@ def lambda_function_or_layer_arn(
     if re.match(pattern, entity_name):
         return entity_name
     if ":" in entity_name:
-        raise Exception('Lambda %s name should not contain a colon ":": %s' % (type, entity_name))
+        client = connect_to_service("lambda")
+        entity_name, alias = entity_name.split(":")
```

Comment:
Thanks @pinzon for looking into this! 👍 

Wondering if it's possible that `entity_name` may contain multiple colons (`:`)..? (although the case of ARNs should technically already be covered in line 586).

We could change this to the following, to be on the very safe side (to avoid `ValueError: too many values to unpack`):
```
entity_name, _, alias = entity_name.rpartition(":")
```